qml/pages/ChatPage.qml

```diff
@@ -681,1172 +681,1307 @@ Page {
         }
     }
 
-    SilicaFlickable {
-        id: chatContainer
-
-        onContentYChanged: {
-            // For some strange reason contentY sometimes is > 0 which doesn't make sense without a PushUpMenu (?)
-            // That leads to the problem that the whole flickable is moved slightly (or sometimes considerably) up
-            // which creates UX issues... As a workaround we are setting it to 0 in such cases.
-            // Better solutions are highly appreciated, contributions always welcome! ;)
-            if (contentY > 0) {
-                contentY = 0;
-            }
+    Drawer {
+        id: messageOptionsDrawer
+
+        property var myMessage: ({})
+        property var userInformation: ({})
+        property var additionalItemsModel: 0
+        property var sourceItem
+
+        signal closeRequested();
+
+        function closeDrawer() {
+            messageOptionsDrawer.closeRequested();
+            messageOptionsDrawer.open = false;
         }
 
         anchors.fill: parent
-        contentHeight: height
-        contentWidth: width
-
-        PullDownMenu {
-            visible: chatInformation.id !== chatPage.myUserId && !stickerPickerLoader.active && !voiceNoteOverlayLoader.active && !messageOverlayLoader.active && !stickerSetOverlayLoader.active
-            MenuItem {
-                id: closeSecretChatMenuItem
-                visible: chatPage.isSecretChat && chatPage.secretChatDetails.state["@type"] !== "secretChatStateClosed"
-                onClicked: {
-                    var secretChatId = chatPage.secretChatDetails.id;
-                    Remorse.popupAction(chatPage, qsTr("Closing chat"), function() { tdLibWrapper.closeSecretChat(secretChatId) });
-                }
-                text: qsTr("Close Chat")
-            }
+        dock: chatPage.isPortrait ? Dock.Bottom : Dock.Right
+        backgroundSize: dock == Dock.Left || dock == Dock.Right ? width / 2 : height / 3
```

Comment:
Spent some time trying to figure how `dock` could become `Dock.Left` but no avail. You probably meant this:
```js
backgroundSize: chatPage.isPortrait ? height / 3 : width / 2
```

Reply:
Oh, I simply copied the content from `Dock` and changed the values. Of course, your proposal makes much more sense...